### PR TITLE
Update Go version to 1.15.3

### PR DESCRIPTION
**Pull request by zrhoffman**
## What does this PR (Pull Request) do?

- This PR makes the Go components of Traffic Control build using Go version 1.15.3.
- Makes the Go unit tests and Traffic Ops API tests run in GitHub Actions when the Go version is updated
- - [x] This PR is not related to any Issue

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Grove
- Traffic Control Client (Go)
- Traffic Monitor
- Traffic Ops
- Traffic Ops ORT
- Traffic Stats
- CI tests

## What is the best way to verify this PR?

Run unit tests and API tests. Since this is only a patch-level version update, [the only changes](https://github.com/golang/go/milestone/171?closed=1) were bugfixes. Breaking changes would be unexpected.

## The following criteria are ALL met by this PR

- [x] Existing tests are sufficient, no additional tests necessary
- [x] The documentation only mentions the Go version with minor version precision, no documentation updates necessary.
- [x] The changelog already mention updating to Go 1.15, no additional changelog message necessary.
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: .github/workflows/go.unit.tests.yaml [hunk 1]
```diff
@@ -20,6 +20,7 @@ name: Go Unit Tests
 on:
   push:
     paths:
+      - GO_VERSION
```
Comment (shamrickus) [inline .github/workflows/go.unit.tests.yaml:23]: Shouldn't this also be added to the go-fmt action?

Comment (zrhoffman) [reply .github/workflows/go.unit.tests.yaml:23]: Yes. Added in 7c14c75271

**Review by shamrickus — APPROVED** [review]
LGTM
ciab-build failed, but not because of this change

**Comment by zrhoffman** [comment]
Ehh queued ciab-build to re-run anyway for the precious :heavy_check_mark:

**Comment by zrhoffman** [comment]
We got the :heavy_check_mark:

**Pull request merged**